src/main/java/com/forgeessentials/api/UserIdent.java

```diff
@@ -489,19 +615,20 @@ public static UserIdent fromString(String string)
         if (string.charAt(0) != '(' || string.charAt(string.length() - 1) != ')' || string.indexOf('|') < 0)
             throw new IllegalArgumentException("UserIdent string needs to be in the format \"(<uuid>|<username>)\"");
         String[] parts = string.substring(1, string.length() - 1).split("\\|", 2);
+        UUID id = null;
+        String str = parts[1] != null && !parts[1].isEmpty() ? parts[1] : null;
         try
         {
-            return get(UUID.fromString(parts[0]), parts[1]);
+            id = UUID.fromString(parts[0]);
         }
         catch (IllegalArgumentException e)
-        {
-            return get((UUID) null, parts[1]);
-        }
+        {}
+        return get(id,str,false,true);
     }
 
     public String toSerializeString()
     {
-        return "(" + (uuid == null ? "" : uuid.toString()) + "|" + username + ")";
+        return "(" + (uuid == null ? "" : uuid.toString()) + "|" + (username == null ? "" : username) + ")";
```

Comment:
Now I know, why we sometimes got `123-123-123-123|null` entries for users :sweat_smile: